### PR TITLE
Reusable resource tag

### PR DESCRIPTION
**Pull request by AdamGordonNY**
also added a few other icons - i got a little confused , I was trying to extend the Badge from Shadcn-ui , and import the icons from lucide , but it wasn't working as I wanted it to.  After reworking it a whole bunch of times, I was able to get the resource tag that also works for the right sidebar/other tags made, let me know what you think

### CODE REVIEW

File: components/shared/ResourceTag.tsx [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+interface CustomBadgeProps extends React.PropsWithChildren {
+  type: ResourceTagType;
+  text?: string;
+}
+
+const ResourceTag = (props: CustomBadgeProps) => {
+  const { children, type, text } = props;
+  const badgeType = type || "plain";
+  const baseClass =
+    "inline-flex px-0.5 py-1 gap-1 rounded-[3px] items-center justify-center rounded-[3px]";
```
Comment (brandonetter) [inline components/shared/ResourceTag.tsx]:
```suggestion
    "inline-flex px-0.5 py-0.5 gap-1 rounded-[3px] items-center justify-center rounded-[3px]";
```
The y padding should be 2px

File: components/shared/ResourceTag.tsx [hunk 2]
```diff
@@ -0,0 +1,56 @@
+import React from "react";
+import workflowBadge from "@/public/icons/monitor.svg";
+import knowledgeBadge from "@/public/icons/greenbubble.svg";
+import componentBadge from "@/public/icons/numberlist.svg";
+import Image from "next/image";
+type ResourceTagType = "knowledge" | "component" | "workflow" | "plain";
+interface CustomBadgeProps extends React.PropsWithChildren {
+  type: ResourceTagType;
```
Comment (brandonetter) [inline components/shared/ResourceTag.tsx]:
```suggestion
  type?: ResourceTagType;
```
If we make this optional, we can short-circuit the value in our class and have a "default" class if you don't want to set that value.

File: components/shared/ResourceTag.tsx [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  const iconSrc = {
+    component: componentBadge,
+    workflow: workflowBadge,
+    knowledge: knowledgeBadge,
+    plain: null,
+  };
+
+  return (
+    <span
+      className={`${baseClass} ${color[type]} hover:${color[type]}`}
```
Comment (brandonetter) [inline components/shared/ResourceTag.tsx]:
```suggestion
      className={`${baseClass} ${color[type || "plain"]} hover:${color[type || "plain"]}`}
```
This is the short-circuit mentioned before. A useful pattern to simplify the developer experience when making things like this. We can say something is `optional` and then provide the default.

We could alternatively provide a default value in our props:
```js

const ResourceTag = ({
  children,
  type = "plain",
  text,
  ...props
}: CustomBadgeProps) => {
```

Either one should be fine- but I think it makes your usage easier.

**Review by brandonetter — CHANGES_REQUESTED** [review]

**Comment by brandonetter** [comment]
Overall this looks really good. Very clean. Nice work.

File: components/shared/ResourceTag.tsx (hunk 3 above)
Comment (AdamGordonNY) [reply components/shared/ResourceTag.tsx]: noted ty

**Review by brandonetter — APPROVED** [review]

**Comment by AdamGordonNY** [comment]
Thank You :)

**Pull request merged**